src/modules/job-info/update.c

```diff
@@ -143,6 +143,7 @@ static void eventlog_continuation (flux_future_t *f, void *arg)
     json_t *context = NULL;
     const char *errmsg = NULL;
     const flux_msg_t *msg;
+    bool skip_respond_error = false;
 
```

Comment:
Same comment on this one.   Just call `eventlog_watch_cancel()` at each `goto error_cancel` site and `goto error` or `goto cleanup` from there as appropriate